### ISSUE TITLE
Headers Being Dropped In Restify

### ISSUE THREAD

**Issue by svozza** [kwhitley/apicache#60]
With Restify 4.3 I've encountered an issue with headers being dropped when a response is being returned from the cache. I've localised the issue to here:

```js
function sendCachedResponse(response, cacheObject) {
    Object.assign(response._headers || {}, cacheObject.headers || {}, {
      'apicache-store': globalOptions.redisClient ? 'redis' : 'memory',
      'apicache-version': pkg.version
    })

    // unstringify buffers
    var data = cacheObject.data
    if (data && data.type === 'Buffer') {
      data = new Buffer(data.data)
    }

    response.writeHead(cacheObject.status || 200, response._headers)

    return response.end(data, cacheObject.encoding)
  }
```

When `response._headers` is `null` the code falls back to `{}` but because the result of `Object.assign` is not saved in a variable it gets dropped silently. The reason why we don't see this in Express is because it always adds the `X-Powered-By: 'Express'` header so `res._headers` is never empty (although anyone who disables this feature will see this problem).

The solution is very simple and I'd be more than happy to raise a PR for this:

```js
function sendCachedResponse(response, cacheObject) {
    // Should the res object be mutated here rather than cloned as I've done?
    const headers = Object.assign({}. response._headers || {}, cacheObject.headers || {}, {
      'apicache-store': globalOptions.redisClient ? 'redis' : 'memory',
      'apicache-version': pkg.version
    })
    // ...
    response.writeHead(cacheObject.status || 200, headers)
    // ...
  }
```

**Comment by kwhitley** [comment]
If that fixes it, by all means add a PR and send it over!  Thanks @svozza 

Please do verify that `const` is valid in Node 4.0 though, as I'm trying to maintain compatibility to 4 still.  I believe it is, I just haven't checked the green sheet recently.  Full ES6 rewrite is coming soon!

**Comment by svozza** [comment]
Thanks for the quick response! `const` is fine in Node 4 (we use it in AWS Lambda all the time) but I decided to mutate the `res` object in the end because I'm not sure if others would be relying on `_headers` being present and containing all the headers.

**Comment by kwhitley** [comment]
Fixed in #61
